src/main/java/login/HttpCookie.java

```diff
@@ -0,0 +1,45 @@
+package login;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class HttpCookie {
+    private static final String COOKIE_PARSER_TOKEN = ";";
+    private static final String COOKIE_KEY_VALUE_APPEND_TOKEN = "=";
+
+    private final Map<String, String> cookies;
+
+    public HttpCookie() {
+        this.cookies = new HashMap<>();
+    }
+
+    public HttpCookie(String cookieString) {
+        if (cookieString == null || cookieString.isBlank()) {
+            this.cookies = Collections.emptyMap();
+            return;
+        }
+
+        this.cookies = Arrays.stream(cookieString.split(COOKIE_PARSER_TOKEN))
+            .map(String::trim)
+            .map(cookie -> cookie.split(COOKIE_KEY_VALUE_APPEND_TOKEN, 2))
+            .collect(Collectors.toMap(token -> token[0], token -> token[1]));
+    }
```

Comment:
생성자 대신에 정적 팩토리 메서드를 활용해보면 어떨까요? 🙂

```suggestion
    public HttpCookie(Map<String, String> cookies) {
        this.cookies = cookies;
    }

    public static HttpCookie ofCookieString(String cookieString) {
        if (cookieString == null || cookieString.isBlank()) {
            return new HttpCookie(Collections.emptyMap());
        }

        var cookies = Arrays.stream(cookieString.split(COOKIE_PARSER_TOKEN))
            .map(String::trim)
            .map(cookie -> cookie.split(COOKIE_KEY_VALUE_APPEND_TOKEN, 2))
            .collect(Collectors.toMap(token -> token[0], token -> token[1]));
        return new HttpCookie(cookies);
    }
```